### PR TITLE
Fix for deployment workflows

### PR DESCRIPTION
**Pull request by eminalemdar**
## Description of the change

Updated deployment workflows to fix error for reference to the deployment id in second job

## Type of change

- [X] Bug fix (non-breaking change that fixes an issue);
- [ ] New feature (non-breaking change that adds functionality);
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected);
- [ ] Other (miscellaneous, GitHub workflow changes, changes to the PR template);

### Development

- [X] Lint rules pass locally;
- [X] All tests related to the changed code pass in development;

### Code review

- [ ] This pull request has a descriptive title and sufficient context for a reviewer. There may be a screenshot or screencast attached;
- [ ] This pull request is no longer a draft;

### Deployment

- [ ] Selected merge strategy is [squash merge](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits);
- [ ] Changes have been reviewed by at least one other engineer;

### CODE REVIEW

File: .github/workflows/preprod-deployment.yml [hunk 1]
```diff
@@ -164,4 +165,4 @@ jobs:
           token: "${{ github.token }}"
           target_url: https://downloads.spacelift.dev/spacelift-vcs-agent
           state: "failure"
-          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
+          deployment_id: ${{ needs.preprod-agent-deployment.outputs.deployment_id }}
```
Comment (adamconnelly) [inline .github/workflows/preprod-deployment.yml:168]: Just wondering, if the build job fails, will we mark the deployment as failed? If not, I think we've got two options:

1. (the simple option) Move the "Create GitHub deployment" step into the upload job.
2. Add a failure task to the build job as well.

I think here we could get away with option 1 on the grounds that you could argue the deployment hasn't started until we begin to upload the binary/image, but honestly I think either is fine here.

Comment (eminalemdar) [reply .github/workflows/preprod-deployment.yml:168]: I think I can just add an extra failure step (like the one in the upload job) to the build job as well. WDYT?

Comment (adamconnelly) [reply .github/workflows/preprod-deployment.yml:168]: Yeah - that should be fine.

Comment (eminalemdar) [reply .github/workflows/preprod-deployment.yml:177]: Added the extra step to both preprod and prod deployment workflows.

File: .github/workflows/prod-deployment.yml [hunk 2]
```diff
@@ -84,6 +84,15 @@ jobs:
           path: build/
           retention-days: 1
 
+      - name: Update deployment status (failure)
+        uses: chrnorm/deployment-status@releases/v1
+        if: failure() && ${{ github.ref == 'refs/heads/production' }}
+        with:
+          token: "${{ github.token }}"
+          target_url: https://downloads.spacelift.dev/spacelift-vcs-agent
```
[the remark this answers is not in the document]
Comment (adamconnelly) [reply .github/workflows/prod-deployment.yml:92]: Yes - if you take a look at the other deployment tasks in each workflow you'll see the URLs are different between prod and preprod.

**Review by adamconnelly — APPROVED** [review]

**Pull request merged**